src/jdk.httpserver/share/classes/com/sun/net/httpserver/Filter.java

```diff
@@ -140,4 +142,117 @@ public abstract void doFilter (HttpExchange exchange, Chain chain)
+     *     server.start();
+     * }</pre>
+     *
+     * @param description the description of the returned filter
+     * @param operation the operation of the returned filter
+     * @return a filter
+     * @throws NullPointerException if any argument is null
+     * @since 17
+     */
+    public static Filter beforeHandler(String description,
+                                       Consumer<HttpExchange> operation) {
+        Objects.requireNonNull(description);
+        Objects.requireNonNull(operation);
+        return new Filter() {
+            @Override
+            public void doFilter(HttpExchange exchange, Chain chain) throws IOException {
+                operation.accept(exchange);
+                chain.doFilter(exchange);
+            }
+            @Override
+            public String description() {
+                return description;
+            }
+        };
+    }
+
+    /**
+     * Returns a post-processing {@code Filter} with the given description and
+     * operation.
+     *
+     * <p>The {@code description} string describes the returned filter. The
+     * {@link Consumer operation} is the effective implementation of the filter.
+     * It is executed for each {@code HttpExchange} after invoking either the
+     * next filter in the chain or the exchange handler (if this filter is the
+     * final filter in the chain).
+     *
+     * @apiNote
+     * An afterHandler filter is typically used to examine the exchange state
+     * rather than modifying it. The filter operation is executed after
+     * {@link Filter.Chain#doFilter(HttpExchange)} is invoked, this means any
+     * subsequent filters in the chain and the exchange handler have been
+     * executed and the response has commonly been sent. The filter is only
+     * expected to send the response in the uncommon case that the exchange
+     * has not been handled before the filter is executed.
+     *
+     * <p> Example of logging the response code of all exchanges:
+     * <pre>{@code
+     *     var filter = Filter.afterHandler("Log response code", (var e) -> log(e.getResponseCode());
+     *     var server = HttpServer.create(new InetSocketAddress(0), 10);
+     *     var context = server.createContext("/", new SomeHandler());
+     *     context.getFilters().add(filter);
+     *     server.start();
+     * }</pre>
+     *
+     * <p> Example of adding a sequence of afterHandler filters to a context
+     * (note the ordering):
+     * <pre>{@code
+     *     var filter1 = Filter.afterHandler("Set a1", (var e) -> e.setAttribute("a1", "some value"));
+     *     var filter2 = Filter.afterHandler("Get a1", (var e) -> doSomething(e.getAttribute("a1")));
+     *     var server = HttpServer.create(new InetSocketAddress(0), 10);
+     *     var context = server.createContext("/", new SomeHandler());
+     *     context.getFilters().add(filter2);
+     *     context.getFilters().add(filter1);
+     *     server.start();
+     * }</pre>
```

Comment:
I don't think "note the ordering" is sufficiently explanatory here: I feel some better description of the sequence of calls that explain that filter2's operation will be executed after filter1's if registered before would be in order. Maybe something like:

```
* <p> Example of adding a sequence of afterHandler filters to a context.<br>
* Because the filter operation is invoked after the next filter in the chain has been invoked,
* the operation of the first after filter registered will be invoked <b>after</b> the operation
* of the second after filter registered (note the ordering):  
* <pre>...</pre>
* <p>The operation of {@code filter2} which has been registered before {@code filter1}
* will be invoked after {@code filter1} operation has been invoked.
```

Maybe renaming `filter1` into `a1Get` and `filter2` into `a1Set` would be less confusing too.